File: 2025/day05/Python/part1/day05pt1.py
def main():
    with open("input.txt", "r", encoding="utf-8") as file:
        lines = []
        for line in file:
            lines.append(line.strip())

        ranges = []
        ids = []

        toggle = True

        for line in lines:
            if line == "":
                toggle = False
                continue
                
            if toggle:
                split = line.split("-")
                front = int(split[0])
                rear = int(split[-1])
                ranges.append([front, rear])
            
            else:
                ids.append(int(line))

        res = 0
        for i in ids:
            for r in ranges:
                if r[0] <= i <= r[-1]:
                    res += 1
                    break # don't over count ids that fall into multiple ranges
            
    print(res)


main()

File: 2025/day05/Go/day05pt1.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func main() {
	file, err := os.Open("input.txt")
	if err != nil {
		panic(err)
		return
	}

	var lines []string
	scanner := bufio.NewScanner(file)
	for scanner.Scan() {
		line := strings.TrimSpace(scanner.Text())
		lines = append(lines, line)
	}

	var ranges [][]int
	var ids []int

	toggle := true

	for _, line := range lines {
		if line == "" {
			toggle = false
			continue
		}

		if toggle {
			parts := strings.Split(line, "-")
			front, _ := strconv.Atoi(parts[0])
			rear, _ := strconv.Atoi(parts[1])
			ranges = append(ranges, []int{front, rear})
		} else {
			value, _ := strconv.Atoi(line)
			ids = append(ids, value)
		}
	}

	res := 0
	for _, id := range ids {
		for _, r := range ranges {
			if r[0] <= id && id <= r[1] {
				res++
				break // do not overcount
			}
		}
	}

	fmt.Println(res)
}